csharp/src/Ice/Connection.cs

```diff
@@ -308,6 +308,10 @@ async Task PerformGoAwayAsync(Exception exception)
                                                                exception.Message,
                                                                cancel).ConfigureAwait(false);
 
+                    // Make sure to yield to release the mutex. It's important to not hold the mutex because the
```

Comment:
It would be nicer to release the mutex explicitly, without this yield.

Reply:
We use this Yield pattern in few places, it has the advantage of being "exception" safe. How do you suggest to do this explicit release? Using the Mutex class and the ReleaseMutex method? I find this a bit error prone. Another option could be to re-arrange this code to no longer rely on a task + mutex but instead use a semaphore and WaitAsync to ensure there's only a single async task to perform the closure... I'm not sure it will be simpler in the end. I can try but I'd rather do this with another PR.